src/sysdolphin/baselib/fog.c

```diff
@@ -0,0 +1,256 @@
+#include <sysdolphin/baselib/fog.h>
+
+#include <dolphin/gx/GX_unknown_001.h>
+#include <sysdolphin/baselib/cobj.h>
+
+static void FogInfoInit(void);
+static void FogAdjInfoInit(void);
+HSD_FogInfo hsdFog = { FogInfoInit };
+HSD_ClassInfo hsdFogAdj = { FogAdjInfoInit };
+
+typedef struct _GXFogAdjTbl {
+	u16 r[10];
+} GXFogAdjTbl;
+
+const GXColor lbl_804DE6F0 = { 0 };
+
+void HSD_FogSet(HSD_Fog* fog)
+{
+    GXFogAdjTbl sp88;
```

Comment:
Let's get these renamed from autogenerated names for the sake of cleanup. I have the actual names labeled in FRAY.